src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs

```diff
@@ -255,7 +255,7 @@ internal static HashSet<string> GetValidSources(IEnumerable<string>? sources, IM
             // It is possible that runtime provider sent relative source path for remote scenario.
             string src = !Path.IsPathRooted(source) ? Path.Combine(Directory.GetCurrentDirectory(), source) : source;
 
-            if (!File.Exists(src))
+            if (!File.Exists(src) && !Directory.Exists(src))
```

Comment:
Note: This is the crux of the change. The rest of the PR is basically plumbing for the new `[DirectoryBasedTestDiscoverer]` attribute.